examples/render/quad_render/main.rs

```diff
@@ -238,38 +170,28 @@ fn main() {
         }
     }
 
-    let image = context.mut_device().create_image::<ColorFormat, _>(
+    let (image, token) = device.create_image::<ColorFormat, _>(
         &mut data,
         gfx::image::TRANSFER_DST | gfx::image::SAMPLED,
         kind,
         1,
     ).unwrap();
+    init_tokens.push(token);
 
-    let image_srv = context.mut_device()
-        .view_image_as_shader_resource(&image)
+    let image_srv = device.view_image_as_shader_resource(&image)
         .unwrap();
 
-    let sampler = context.mut_device().create_sampler(
+    let sampler = device.create_sampler(
         i::SamplerInfo::new(
             i::FilterMethod::Bilinear,
             i::WrapMode::Clamp,
         )
     );
 
-    device.update_descriptor_sets(&[
-        pso::DescriptorSetWrite {
-            set: &set0[0],
-            binding: 0,
-            array_offset: 0,
-            write: pso::DescriptorWrite::SampledImage(vec![(image_srv.resource(), i::ImageLayout::Undefined)]),
-        },
-        pso::DescriptorSetWrite {
-            set: &set1[0],
-            binding: 0,
-            array_offset: 0,
-            write: pso::DescriptorWrite::Sampler(vec![sampler.resource()]),
-        },
-    ]);
+    device.update_descriptor_sets()
+        .write(desc_data.sampled_image(&desc), 0, &[&image_srv as _])
```

Comment:
I wonder if we can get rid of those `X as _` things?

Reply:
Unfortunately type inference does not seem to go through without it.